File: D.S.A/strings/remove-outermost-parenthesis.js
/**
 * @param {string} s
 * @return {string}
 */
var removeOuterParentheses = function(s) {
   
   let counter=0;
   let ans='';

   for(let i=0;i<s.length;i++){

    if(counter===0){
        if(s[i]==='('){
            counter++;
        }
    }else{
      if(s[i]==='('){
            counter++;
            ans+=s[i];
        }else{
            counter--;
            if(counter!==0){
                ans+=s[i];
            }
        }
        
    }
   }

    return ans;
};
// Explanation:
// - Track nesting depth with `counter`.
// - Skip adding opening bracket when depth is 0.
// - Add inner opening brackets and increase depth.
// - On closing bracket, decrease depth first.
// - Add closing bracket only if still inside primitive.

File: Java/dsa/arrays/easy/strings/RemoveOutermostParenthesis.java
public class RemoveOutermostParenthesis {
    public static String removeOuterParentheses(String s) {
        int counter = 0;
        StringBuilder ans = new StringBuilder();

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (ch == '(') {
                if (counter > 0) {
                    ans.append(ch);
                }
                counter++;
            } else {
                counter--;
                if (counter > 0) {
                    ans.append(ch);
                }
            }
        }
        return ans.toString();
    }
}
